src/annotator/highlighter.js

```diff
@@ -161,17 +161,37 @@ function wholeTextNodesInRange(range) {
     return [];
   }
 
-  const textNodes = [];
+  /** @type {Node|null} */
+  let root = range.commonAncestorContainer;
+  if (root.nodeType !== Node.ELEMENT_NODE) {
+    // If the common ancestor is not an element, set it to the parent element to
+    // ensure that the loop below visits any text nodes generated by splitting
+    // the common ancestor.
+    root = root.parentElement;
+  }
+  if (!root) {
+    // If there is no root element then we won't be able to insert highlights,
+    // so exit here.
+    return [];
+  }
 
-  forEachNodeInRange(range, node => {
-    if (node.nodeType !== Node.TEXT_NODE) {
-      return;
+  const textNodes = [];
+  const nodeIter = /** @type {Document} */ (root.ownerDocument).createNodeIterator(
```

Comment:
Might be helpful to add a comment that this `nodeIterator` will _only_ return Text nodes.